src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaTransportControls.MediaPlayer.cs

```diff
@@ -44,10 +46,65 @@ private void BindMediaPlayer()
 			_stopButton.Maybe(p => p.Tapped += Stop);
 			_skipForwardButton.Maybe(p => p.Tapped += SkipForward);
 			_skipBackwardButton.Maybe(p => p.Tapped += SkipBackward);
+			_fastForwardButton.Maybe(p => p.Tapped += ForwardButton);
+			_rewindButton.Maybe(p => p.Tapped += RewindButton);
+			_progressSlider.Maybe(p => p.Tapped += TappedProgressSlider);
+			_progressSlider.Maybe(p => p.DragLeave += DragLeaveProgressSlider);
+			_progressSlider.Maybe(p => p.DragOver += DragOverProgressSlider);
 
 			_subscriptions.Disposable = AttachThumbEventHandlers(_progressSlider);
 		}
-
+		public void TappedProgressSlider(object sender, RoutedEventArgs e)
+		{
+			if (double.IsNaN(_progressSlider.Value))
+			{
+				return;
+			}
+			if (_mediaPlayer != null)
+			{
+				_wasPlaying = _mediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.Buffering || _mediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.Playing;
+				_mediaPlayer.Pause();
+				_mediaPlayer.PlaybackSession.Position = TimeSpan.FromSeconds(_progressSlider.Value);
+				if (_wasPlaying)
+				{
+					_mediaPlayer.Play();
+				}
+			}
+		}
+		public void DragLeaveProgressSlider(object sender, RoutedEventArgs e)
+		{
+			if (double.IsNaN(_progressSlider.Value))
+			{
+				return;
+			}
+			if (_mediaPlayer != null)
+			{
+				_wasPlaying = _mediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.Buffering || _mediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.Playing;
+				_mediaPlayer.Pause();
+				_mediaPlayer.PlaybackSession.Position = TimeSpan.FromSeconds(_progressSlider.Value);
+				if (_wasPlaying)
+				{
+					_mediaPlayer.Play();
+				}
+			}
+		}
+		public void DragOverProgressSlider(object sender, RoutedEventArgs e)
+		{
```

Comment:
Should we be adjusting play position whilst they're dragging? I would have assumed once they release the slider we set play position